subscriptions-api.md

```diff
@@ -0,0 +1,613 @@
+implementation. If such a separate API endpoint exists and creating
+subscriptions is not feasible on such a separate API endpoint, the respective
+operation might not be available. For instance, creating an MQTT subscription is
+not feasible from outside an MQTT connection, but an existing subscription may
+be enumerable through a subscription collection and deleting the subscription
+from the collection might terminate it.
+
+#### 3.2.1. Subscription Object
+
+A subscription manager manages a collection of subscriptions. The upper limit on
+how many subscriptions are supported is implementation specific.
+
+Each subscription is represented by an object that has the following properties:
+
+- **id** (string) – REQUIRED. The unique identifier of the subscription in the
+  scope of the subscription manager.
+- **protocol** (string) - REQUIRED. Identifier of a delivery protocol. Because of
+  WebSocket tunneling options for AMQP, MQTT and other protocols, the URI
+  scheme is not sufficient to identify the protocol. The protocols with
+  existing CloudEvents bindings are identified as "AMQP", "MQTT3", "MQTT5",
+  "HTTP", "KAFKA", and "NATS". An implementation MAY add support for further
+  protocols.
+- **protocolsettings** (map) - OPTIONAL. A set of settings specific to the
+  selected delivery protocol provider. Options for those settings are listed
+  in the following subsection. An implementation can offer more options.
+  Examples for such settings are credentials, which generally vary by
+  transport, rate limits and retry policies, or the QoS mode for MQTT.
+  See the [Protocol Settings](#322-protocol-settings) section for further details.
+- **sink** (URI) - REQUIRED. The address to which events shall be delivered using
+  the selected protocol. The format of the address MUST be valid for the
+  selected protocol or one of the protocol's own transport bindings (e.g. AMQP
+  over WebSockets).
+- **filter** - OPTIONAL. A filter is an expression of a particular filter
+  dialect that evaluates to true or false and that determines whether an
+  instance of a cloud event should be delivered to the subscription's sink. If
+  a filter expression evaluates to false, the event MUST NOT be sent to the
+  sink. Support for particular filter dialects might vary across different
+  subscription managers. If a filter dialect is specified in a subscription
+  that is unsupported by the subscription manager, creation or updates of the
+  subscription MUST be rejected with an error. See the [Filter
+  Dialects](#323-filter-dialects) section for further details.
+
+#### 3.2.2 Protocol Settings
+
+This section enumerates protocol-specific delivery options for the
+protocol-settings map, including default values where necessary.
+
+##### 3.2.2.1. HTTP
+
+For HTTP, the following settings properties SHOULD be supported by all
+implementations.
+
+- **proxyurl** (URI) – OPTIONAL. The URI of the proxy server to use for sending
```

Comment:
We might need to say that if a proxyurl is specified then the subscription manager MUST use it when trying to send events - IOW, ignoring it isn't an option.

Reply:
actually, do we need to make similar statements for all of these? Do we need to say what happens if a subscription manage doesn't support it? Or is it truly optional and the consumer has no idea which ones will be adhered to?

Reply:
Good points.

Reply:
I am removing the proxy info since that doesn't belong here, after all. This is not a per-subscription config.